cpp/CMakeLists.txt

```diff
@@ -350,88 +350,6 @@ if(UNIX)
   add_custom_target(iwyu-all ${BUILD_SUPPORT_DIR}/iwyu/iwyu.sh all)
 endif(UNIX)
 
-#
-# Set up various options
-#
-
-if(ARROW_BUILD_BENCHMARKS
-   OR ARROW_BUILD_TESTS
-   OR ARROW_BUILD_INTEGRATION
-   OR ARROW_FUZZING)
-  set(ARROW_TESTING ON)
-endif()
-
-if(ARROW_GANDIVA)
-  set(ARROW_WITH_RE2 ON)
-  set(ARROW_WITH_UTF8PROC ON)
-endif()
-
-if(ARROW_BUILD_INTEGRATION AND ARROW_FLIGHT)
-  set(ARROW_FLIGHT_SQL ON)
-endif()
-
-if(ARROW_FLIGHT_SQL)
-  set(ARROW_FLIGHT ON)
-endif()
-
-if(ARROW_CUDA
-   OR ARROW_FLIGHT
-   OR ARROW_PARQUET
-   OR ARROW_BUILD_TESTS
-   OR ARROW_BUILD_BENCHMARKS)
-  set(ARROW_IPC ON)
-endif()
-
-if(ARROW_SUBSTRAIT)
-  set(ARROW_PARQUET ON)
-  set(ARROW_IPC ON)
-  set(ARROW_COMPUTE ON)
-  set(ARROW_DATASET ON)
-endif()
-
-if(ARROW_SKYHOOK)
-  set(ARROW_DATASET ON)
-  set(ARROW_PARQUET ON)
-  set(ARROW_WITH_LZ4 ON)
-  set(ARROW_WITH_SNAPPY ON)
-endif()
-
-if(ARROW_TESTING)
-  set(ARROW_JSON ON)
-endif()
-
-if(ARROW_DATASET)
-  set(ARROW_COMPUTE ON)
-  set(ARROW_FILESYSTEM ON)
-endif()
-
-if(ARROW_PARQUET)
-  set(ARROW_COMPUTE ON)
-endif()
-
-if(ARROW_PYTHON)
-  set(ARROW_COMPUTE ON)
-  set(ARROW_CSV ON)
-  set(ARROW_DATASET ON)
-  set(ARROW_FILESYSTEM ON)
-  set(ARROW_HDFS ON)
-  set(ARROW_JSON ON)
-endif()
```

Comment:
FWIW this broke my local development because of no longer including those (although I should probably start using presets ..)

Now, it's probably fine to remove this now Python C++ has moved, but we _do_ assume that some C++ modules are built on the pyarrow side (eg we assume that CSV is always built, while with the above change you need to ensure manually that this is done in your cmake call). 
In any case we should update the documentation at https://arrow.apache.org/docs/dev/developers/python.html#build-and-test to indicate that there are a few components required to be able to build pyarrow.

Reply:
Ah, thanks for the info.

OK. I add `ARROW_PYTHON` again but I make it a deprecated option. We'll remove `ARROW_PYTHON` eventually.

See also: ARROW-17868